---
title: "Tumor-based Benchmark"
output: md_document
vignette: >
  %\VignetteIndexEntry{tumorBench}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Tumor-based benchmark
## Introduction
This benchmark is based on the cptac dataset which contains genomics, proteomics
and phosphoproteomics information of multiple patients from different cancer types. 
Here, the proteomics data was used to identify a gold standard set of deregulated 
kinases per patient which are expected to have an increased or decreased activity.

The phosphoproteomic data is then used to infer kinase activities and to evaluate
which method and kinase-substrate library is able to recapitulate the gold standard
set of kinases.

For this, we provide the normalized abundance of phosphorylation sites for each
patient across different cancer types.

## Getting set up
We first load the required packages to run the benchmark.

```{r setup}
library(benchmarKIN)
library(dplyr)
library(tibble)
```

## CPTAC data
We then take a look at the phosphoproteomics data from CPTAC. This data is 
stored as a long data frame in cptacData. For our analysis we will transform
it to a list of wide data frames for each cancer type which can be used for
activity inference.

```{r data}
cptac_list <- purrr::map(unique(cptacData$cancer), function(cancer_type){
  cptacData %>%
    dplyr::filter(cancer == cancer_type) %>%
    tidyr::pivot_wider(names_from = "sample", values_from = "statistic", values_fill = NA) %>%
    dplyr::select(-cancer) %>%
    tibble::column_to_rownames("id")
})

names(cptac_list) <- unique(cptacData$cancer)
```

## Activity estimation
Here, we will test the z-score (as implemented by RoKAI) in combination with the
curated kinase-substrate library as in example.

In theory any other method can also be tested with that data.

### Kinase-substrate library
For that we have already processed a version of a combination of three curated libraries,
namely PhosphoSitePlus, PTMsigDB (excluding iKiP-DB) and the gold standard set of GPS 5.0,
that can be mapped to our phosphorylation site ids.

```{r curated}
head(curated)
```
For the tumor-based benchmark we remove auto-phosphorylation sites from the
library to avoid data leakage from the gold standard sets used in this benchmark.

```{r autophos}
curated_filt <- curated %>%
  dplyr::filter(source != target_protein)
```

After mapping the phosphorylation sites we can bring it into the right format to
run the run_zscore function.

```{r map_ppsp}
pps_id <- cptacData$id %>% 
  unique()
map_pps <- data.frame(pps_id = pps_id) %>%
  dplyr::mutate(ENSEMBL = purrr::map_chr(stringr::str_split(pps_id, "\\|"), 1)) %>%
  dplyr::mutate(ENSEMBL = purrr::map_chr(stringr::str_split(ENSEMBL, "\\."), 1))  %>%
  dplyr::mutate(sequence = purrr::map_chr(stringr::str_split(pps_id, "\\|"), 4))
mapped_curated <- curated_filt %>%
  dplyr::left_join(map_pps, by = c("ENSEMBL", "sequence"), relationship = "many-to-many") %>%
  dplyr::mutate(target = pps_id) %>%
  dplyr::filter(!is.na(pps_id)) %>%
  dplyr::select(source, target, mor) %>%
  dplyr::distinct()
```

### Activity inference using the z-score 
The z-score calculates a score for each kinase by aggregating the change in
abundance of the direct targets in relation to changes in the non-targets.

```{r activity}
act_scores <- purrr::map(cptac_list, ~list(zscore_ppsp = run_zscore(mat = .x, network = mapped_curated)))
```

## Tumor-based benchmark
### Protein-based gold standard set selection
For each cancer type, a gold standard set of kinases was developed based on
the protein levels for each kinase. For the gold standard set kinases 
in the top and bottom 5% relative to the normal distribution of the protein levels 
were selected. Here we can look at the kinases which were located in the top
5% and are supposed to have a high activity for patients which BRCA. 
Additionally we also provide the gold standard sets using the top and bottom
2.5%, 10% and 15% (cptacGS_2pt5, cptacGS_10 and cptacGS_15) of kinases relative 
to the normal distribution. 

```{r cptacGS}
data(cptacGS_5)
str(cptacGS_5$BRCA$GS_pos_pairs)
```
### Activating site-based gold standard set selection
In addition to the protein-based selection,  we developed an alternative gold 
standard set of kinases using the activating sites on kinases. Hereby, the same 
thresholds were applied to the kinase phosphosite levels. 
This gold standard set can also be accessed and used in the 
benchmark (cptacGS_act_2pt5, cptacGS_act_5, cptacGS_act_10 and cptacGS_act_15).

```{r cptacGSact}
data(cptacGS_act_5)
str(cptacGS_act_5$BRCA$GS_pos_pairs)
```

### Area under the receiver operator curve
The inferred kinase activities can then be evaluated using the benchmarkROC()
function. Hereby, it is important to note that the kinase symbol needs to match
with the gold standard set (here gene symbols).
```{r benchmark}
performance_prot <- benchmarkROC(score_lists = act_scores, GS_list = cptacGS_5)
performance_act <- benchmarkROC(score_lists = act_scores, GS_list = cptacGS_act_5)
```

We can then plot the areas under the receiver operator curve (AUROCs) and can 
compare this with other methods or prior knowledge resources.
```{r plot}
performance <- rbind(performance_prot %>% tibble::add_column(benchmark = "protein-based"),
                     performance_act %>% tibble::add_column(benchmark = "activating site-based"))
auroc_p <- performance %>%
  ggplot2::ggplot(ggplot2::aes(x=method, y=auroc, fill = benchmark)) +
  ggplot2::geom_boxplot(outlier.size=1, lwd=0.6) +
  ggplot2::geom_hline(yintercept = 0.5)

auroc_p
```
